flink-core/src/main/java/org/apache/flink/api/java/typeutils/PojoTypeInfo.java

```diff
@@ -317,7 +308,39 @@ public int getFieldIndex(String fieldName) {
 
 		return new PojoSerializer<T>(getTypeClass(), fieldSerializers, reflectiveFields, config);
 	}
-	
+
+	@Override
+	@PublicEvolving
+	public <F> FieldAccessor<T, F> getFieldAccessor(String fieldExpression, ExecutionConfig config) {
+		
+		FieldAccessor.FieldExpressionDecomposition decomp = FieldAccessor.decomposeFieldExpression(fieldExpression);
+
+		// get field
+		PojoField field = null;
+		TypeInformation<?> fieldType = null;
+		for (int i = 0; i < fields.length; i++) {
+			if (fields[i].getField().getName().equals(decomp.head)) {
+				field = fields[i];
+				fieldType = fields[i].getTypeInformation();
+				break;
+			}
+		}
+		if (field == null) {
+			throw new InvalidFieldReferenceException("Unable to find field \""+decomp.head+"\" in type "+this+".");
+		}
+
+		if(decomp.tail == null) {
```

Comment:
It would great if you could document that tail can be null in `FieldExpressionDecomposition`.